packages/run-protocol/src/vpool-xyk-amm/addPool.js

```diff
@@ -44,65 +41,54 @@ export const makeAddIssuer = (
       X`${keyword} asset not fungible (must use NAT math)`,
     );
 
-    if (brandToLiquidityIssuer.has(secondaryBrand)) {
-      return brandToLiquidityIssuer.get(secondaryBrand);
-    }
+    /** @type {(brand: Brand) => Promise<ZCFMint>} */
+    const makeLiquidityMint = brand => {
+      assert(
+        !isInSecondaries(brand),
+        X`issuer ${secondaryIssuer} already has a pool`,
+      );
 
-    assert(
-      !isInSecondaries(secondaryBrand),
-      X`issuer ${secondaryIssuer} already has a pool`,
+      const liquidityKeyword = `${keyword}Liquidity`;
+      zcf.assertUniqueKeyword(liquidityKeyword);
+
+      return E.when(
+        Promise.all([
+          zcf.saveIssuer(secondaryIssuer, keyword),
+          zcf.makeZCFMint(liquidityKeyword, AssetKind.NAT, DISPLAY_INFO),
+        ]),
+      )
+        .then(([issuer, mint]) => {
+          console.log(
+            X`Saved issuer ${secondaryIssuer} to keyword ${keyword} and got back ${issuer}`,
+          );
+          return mint;
+        })
+        .catch(e => {
+          console.error(
+            X`Failure Saving issuer ${secondaryIssuer}. Not added to Reserve`,
+          );
+          throw e;
+        });
+    };
+    /** @type {(brand: Brand) => Promise<void>} */
+    const finish = brand => {
+      // defer lookup until necessary. more aligned with governed
+      // param we expect this to be eventually.
+      const addIssuerToReserve = getAddIssuerToReserve();
+
+      // tell the reserve about this brand, which it will validate by
+      // calling back to AMM for the issuer
+      return addIssuerToReserve(brand);
+    };
+
+    const liquidityMint = await brandToLiquidityMintProvider.provideAsync(
```

Comment:
Be consistent about `// AWAIT /////////` markers within a function! There's one on the `await` above, so there must be one here as well.

Reply:
> I can't find any documentation about when and when not to use // AWAIT. Does it just mean that there's an await token on the next line? If so, that seems like a code editing concern, not source code.

Usually on the previous line.    We use it to raise awareness in functions with delicate timing/interaction. That's why I'm asking for consistency within a function.

> If we are going to be consistent, shouldn't we be consistent across the repo?

We've found it to be distracting when overused.

> I'll make the change once I can read documentation on when it's supposed to be used. I'll make the documentation if necessary.

I'd talk with @dtribble

Reply:
For now I removed the `await`.